extensions/smallrye-reactive-messaging-kafka/deployment/pom.xml

```diff
@@ -42,6 +42,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-jackson-deployment</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-hibernate-orm-deployment</artifactId>
+        </dependency>
```

Comment:
I don't think you can add a deployment dependency to an extension without also adding the runtime implementation. And that is going to be an issue.

Looks like this build item should be an SPI module (unfortunately).

Reply:
Ok, but there *are* dependencies to the runtime in the runtime module, they're just optional:

https://github.com/quarkusio/quarkus/blob/ba8c0fef65051e2743800a67f2b92cb5c59fc32c/extensions/smallrye-reactive-messaging-kafka/runtime/pom.xml#L34-L43

I'll move this to SPI, but I wonder if the requirements should be adjusted to take into account optional dependencies.